### PR TITLE
ARM srop support

### PR DESCRIPTION
**Pull request by eQu1NoX**
Temporary usage example here : https://bitbucket.org/eQu1NoX/pwntest/src/23ee885115197c6af544fcb3d41b2cb0d90afb5e/sploit.py?at=master

### CODE REVIEW

File: pwnlib/rop/srop.py [hunk 1]
```diff
@@ -153,6 +169,16 @@ def syscall(self, v):
     def syscall_register(self):
         return ABI.syscall(self.arch).syscall_register
 
+    def fix_offsets(self, objstr):
```
Comment (zachriggle) [inline pwnlib/rop/srop.py]: Any reason we can't extend `registers` to actually contain this information, rather than padding it out?

Comment (zachriggle) [inline pwnlib/rop/srop.py]: The remaining stuff is all here:

http://lxr.free-electrons.com/source/arch/arm/include/asm/ucontext.h#L86 <== aux_sigframe
http://lxr.free-electrons.com/source/arch/arm/include/asm/ucontext.h#L64 <== vpf_sigframe
http://lxr.free-electrons.com/source/arch/arm/include/asm/user.h#L88 <== user_vfp
http://lxr.free-electrons.com/source/arch/arm/include/asm/user.h#L97 <== user_vfp_exc

**Comment by zachriggle** [comment]
Looks like the Travis failure is ephemeral.  Squashing this into a single commit would be great :)

**Comment by eQu1NoX** [comment]
Hi Zach, I've squashed in the commits!

**Comment by zachriggle** [comment]
Looks like the CI failure is my fault, fixed.

**Pull request merged**